### PR TITLE
Fr excel report improvements

### PR DESCRIPTION
**Pull request by Magnitus-**
Fr excel report improvements

### CODE REVIEW

File: client/src/components/screens/PatientVariant/index.jsx [hunk 1]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
-const REPORT_HEADER_NUCLEOTIDIC_VARIATION = 'Variation nucléotidique';
-const REPORT_HEADER_STATUS_PARENTAL_ORIGIN = 'Statut (origine parentale)';
-const REPORT_HEADER_ALLELIC_FREQUENCY = 'Fréquence allélique';
-const REPORT_HEADER_SILICO_PREDICTION = 'Prédiction in silico';
-const REPORT_HEADER_CLIN_VAR = 'ClinVar';
-
 const getValue = curryRight(get)('');
-const valuePresent = x => (x !== undefined && x.trim());
-const join = sep => a => (isArray(a) ? a.join(sep) : '');
+const valuePresent = x => (x !== undefined && x !== '' && x !== null);
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:69]: Or (probably being nitpicky about making it a few characters shorter):

```
const valuePresent = x => !_.isNil(x) && x !== ''
```

File: client/src/components/screens/PatientVariant/index.jsx [hunk 2]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
+  const lines = [variant.mutationId, '\n'];
+
+  const bdExts = has(variant, 'variant.bdExt.dbsnp') ? variant.bdExt.dbsnp : [];
+  lines.push(...insertCR(bdExts));
+
+  const geneLine = [
+    'Gène: ',
+    {
+      bold: true,
+      value: `${getValue('geneSymbol')(gene)}`,
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:89]: You can also write:

```
getValue(gene, 'geneSymbol')
```
Curry allows you to pass as many arguments as you want to. If you know all your arguments on invocation, you don't need to wrap every argument in a separate parenthesis

File: client/src/components/screens/PatientVariant/index.jsx (hunk 1 above)
Comment (FrancisRenaud) [reply client/src/components/screens/PatientVariant/index.jsx:69]: En fait, avec la fonction isNil je n'avais pas le résultat que je voulais. Je vais réessayer pour voir ...

File: client/src/components/screens/PatientVariant/index.jsx [hunk 3]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
+    },
+    {
+      bold: false,
+    },
+    '\n',
+  ];
+
+  lines.push(...geneLine);
+
+  const transcriptsRefSeqIds = c => c.transcripts.filter(isCanonical).map(t => getValue('refSeqId')(t)).filter(valuePresent);
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:99]: See my previous comment about getValue

File: client/src/components/screens/PatientVariant/index.jsx [hunk 4]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
+  const refTranscripts = variant.consequences.filter(valuePresent).map(transcriptsRefSeqIds).filter(valuePresent);
+  lines.push(...insertCR(refTranscripts));
+
+  // For some reason, point form doesn't work in filter for the _ghas function ...
+  const cdnaChanges = variant.consequences.filter(c => _has('cdnaChange', c)).map(c => `c.${getValue('cdnaChange')(c)}`);
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:104]: See my previous comment about getValue

File: client/src/components/screens/PatientVariant/index.jsx [hunk 5]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
+  lines.push(...insertCR(cdnaChanges));
+
+  // Same as above ...
+  const aaChanges = variant.consequences.filter(c => _has('aaChange')(c)).map(c => `p.(${getValue('aaChange')(c)})`);
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:108]: See my previous comment about getValue

File: client/src/components/screens/PatientVariant/index.jsx [hunk 6]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
-const parentalOriginForDonor = d => `${zygosity(d)}\n${coverage(d)}`;
-const parentalOrigin = (variant, _gene) => join(' ')(variant.donors.map(parentalOriginForDonor).filter(valuePresent));
+/**
+ * Cell generator for parental origin column
+ * @param {*} variant
+ * @param {*} gene
+ */
+const parentalOriginLines = (variant, _gene) => {
+  const zygosity = (donor) => {
+    const zygoCode = d => getValue('zygosity')(d);
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:121]: See my previous comment about getValue

File: client/src/components/screens/PatientVariant/index.jsx [hunk 7]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
+    return zygoCode(donor) === 'HOM'
+      ? intl.get('screen.variantDetails.homozygote')
+      : intl.get('screen.variantDetails.homozygote');
+  };
+
+  const coverage = donor => [
+    intl.get('screen.patientvariant.parentalOrigin.variantConverage'),
+    `${getValue('adAlt')(donor)}/${getValue('adTotal')(donor)} ${intl.get('screen.patientvariant.parentalOrigin.sequenceReads')}`,
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:129]: See my previous comment about getValue

File: client/src/components/screens/PatientVariant/index.jsx [hunk 8]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
+ */
 const clinVar = (variant, _gene) => {
   if (!variant.clinvar) {
-    return '';
+    return 0;
   }
 
-  const cvcs = `${getValue('clinvar_clinsig')(variant.clinvar)}, ${getValue('clinvar_id')(variant.clinvar)}`;
-  return cvcs || '';
+  const clinvarLabel = intl.get(`clinvar.value.${getValue('clinvar_clinsig')(variant.clinvar)}`);
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:193]: See my previous comment about getValue

File: client/src/components/screens/PatientVariant/index.jsx [hunk 9]
```diff
@@ -65,99 +65,159 @@ const COLUMN_WIDTHS = {
...
+  const cvcs = `${clinvarLabel} (${intl.get('screen.patientvariant.clinVarVariationId')}: ${getValue('clinvar_id')(variant.clinvar)})`;
```
Comment (Magnitus-) [inline client/src/components/screens/PatientVariant/index.jsx:194]: See my previous comment about getValue

File: client/src/components/screens/PatientVariant/index.jsx (hunk 1 above)
Comment (Magnitus-) [reply client/src/components/screens/PatientVariant/index.jsx:69]:
```
> const valuePresent = x => !_.isNil(x) && x !== ''
undefined
> valuePresent('')
false
> valuePresent('a')
true
> valuePresent(null)
false
> valuePresent(1)
true
```

Comment (Magnitus-) [reply client/src/components/screens/PatientVariant/index.jsx:69]: Pour lodash 4.17.17 sur node 14 anyways

Comment (FrancisRenaud) [reply client/src/components/screens/PatientVariant/index.jsx]: Oui, c'est bon, j'ai fait des tests aussi!

**Pull request merged**